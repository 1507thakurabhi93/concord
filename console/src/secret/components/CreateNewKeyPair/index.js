import React from 'react';
import { Field, reduxForm } from 'redux-form';

import { Form, Card, Button, Segment } from 'semantic-ui-react';

const CreateNewKeyPair = props => {
  const { handleSubmit, pristine, reset, submitting } = props;
  return (
    <Segment>
      <h2>Create New Key Pair</h2>
      <form onSubmit={ handleSubmit }>

            <div>
                <label>Concord ID</label>
                <Field
                    name="ConcordId"
                    component={Form.Input}
                    type="text"
                    placeholder="Concord ID"
                />
            </div>

            {/* <div>
                <label>Generate Password</label>
                <Field
                    name="generatePassword"
                    component={Form.Checkbox}
                    type="checkbox"
                />
            </div> */}

            {/* <div>
                <label>Store password</label>
                <Field
                    name="storePassword"
                    component={Form.Input}
                    type="password"
                    placeholder="Store Password"
                />
            </div> */}

            <div>
                <Button type="submit" disabled={pristine || submitting}>Submit</Button>
            </div>

      </form>
    </Segment>
  );
};

export default reduxForm({
  form: 'CreateNewKeyPair',
})(CreateNewKeyPair);
