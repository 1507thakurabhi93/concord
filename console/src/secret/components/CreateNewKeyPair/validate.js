const validate = values => {
    const errors = {};
    if (!values.concordId) {
      errors.concordId = 'Required';
    }
    return errors;
  };
  
  export default validate;
  