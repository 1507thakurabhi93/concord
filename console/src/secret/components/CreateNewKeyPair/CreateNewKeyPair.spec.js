// @flow

import React from 'react';
import { shallow, mount } from 'enzyme';

import CreateNewKeyPair from './';

describe('CreateNewKeyPair', () => {

    it('should render without crashing', () => {
        shallow(<CreateNewKeyPair success={false} error={false} message={" "} />);
    });

    it('should match snapshots for various states of success and error', () => {
        const successStates = [ true, false ];
        const errorStates = [ true, false ];
        
        successStates.forEach( s => {
            errorStates.forEach( e => {
                let wrapper = shallow(<CreateNewKeyPair success={s} error={e} />)
                expect( wrapper ).toMatchSnapshot();
            });
        });
    });

})