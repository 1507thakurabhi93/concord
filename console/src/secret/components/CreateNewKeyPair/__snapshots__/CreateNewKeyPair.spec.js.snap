// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewKeyPair should match snapshots for various states of success and error 1`] = `
<Connect(Form(CreateNewKeyPair))
  destroyOnUnmount={true}
  enableReinitialize={false}
  error={true}
  forceUnregisterOnUnmount={false}
  form="CreateNewKeyPair"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  success={true}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;

exports[`CreateNewKeyPair should match snapshots for various states of success and error 2`] = `
<Connect(Form(CreateNewKeyPair))
  destroyOnUnmount={true}
  enableReinitialize={false}
  error={false}
  forceUnregisterOnUnmount={false}
  form="CreateNewKeyPair"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  success={true}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;

exports[`CreateNewKeyPair should match snapshots for various states of success and error 3`] = `
<Connect(Form(CreateNewKeyPair))
  destroyOnUnmount={true}
  enableReinitialize={false}
  error={true}
  forceUnregisterOnUnmount={false}
  form="CreateNewKeyPair"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  success={false}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;

exports[`CreateNewKeyPair should match snapshots for various states of success and error 4`] = `
<Connect(Form(CreateNewKeyPair))
  destroyOnUnmount={true}
  enableReinitialize={false}
  error={false}
  forceUnregisterOnUnmount={false}
  form="CreateNewKeyPair"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  success={false}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
